airflow/api_fastapi/routes/public/dag_run.py

```diff
@@ -57,3 +57,19 @@ async def delete_dag_run(dag_id: str, dag_run_id: str, session: Annotated[Sessio
         )
 
     session.delete(dag_run)
+
+
+@dag_run_router.patch("/{dag_run_id}", responses=create_openapi_http_exception_doc([400, 401, 403, 404]))
+async def modify_dag_run(
+    dag_id: str, dag_run_id: str, state: DAGRunPatchBody, session: Annotated[Session, Depends(get_session)]
+) -> DAGRunResponse:
+    """Modify a DAG Run."""
+    dag_run = session.scalar(select(DagRun).filter_by(dag_id=dag_id, run_id=dag_run_id))
+    if dag_run is None:
+        raise HTTPException(
+            404, f"The DagRun with dag_id: `{dag_id}` and run_id: `{dag_run_id}` was not found"
+        )
+
+    setattr(dag_run, "state", state.state)
+
```

Comment:
Cf the legacy implementation. Setting a DagRun state has side effect and should be using the utility functions `set_dag_run_state_to_success`, `set_dag_run_state_to_queued`, etc... We cannot just set the `DagRun.state` value in the db.

Also we should also use the DagBag for consistency

Reply:
@pierrejeambrun , sorry. What do you mean by "Cf"? You also mentioned "Cf" in a previous comment and I don't know what it means.

I will refer to the legacy code and update this. Thanks

Reply:
"Cf" is for 'confer' / 'refer to' / 'compare'

Reply:
@pierrejeambrun ,

>  Also we should also use the DagBag for consistency
I'm using dagbag to get the dag now. But, I'm trying to understand why we do this. Is this because, the data in DB could be stale and the DagBag always will have the latest data, from the most recent parse?

Reply:
I just updated it according to legacy implementation. Thanks!

Reply:
I think we should always take from the DB.

The only exception is when the DagBag dag contains more informations that we want to return. (This can happens because the dagbag has additionnal attributes). Otherwise if everything is self contained from the DB we should only read from the DB.

Reply:
@pierrejeambrun , Is the current implementation all right?  Or Do I need to make any changes here?